validation/src/main/scala/hmda/validation/rules/EditCheck.scala

```diff
@@ -0,0 +1,10 @@
+package hmda.validation.rules
+
+import hmda.validation.dsl.{ CommonDsl, Result }
+
+abstract class EditCheck[T] extends CommonDsl {
+
+  def name: String
+
+  def apply(lar: T): Result
```

Comment:
Done. (I went with `input` for lack of another idea, but would take suggestions.)

Do you have thoughts on where this class should live? I thought about putting it in the same package with `ValidationApi` (since yes, I agree that the function currently called `doCheck` will end up there), but then it felt weird to have that package refer to the DSL, and... well, that's as far as I got with thinking about it, as I decided to focus first on the class-level stuff.